### PR TITLE
install pnpm only first time or on version mismatch

### PR DESCRIPTION
**Pull request by starbirdtech383**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [x] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Comment by starbirdtech383** [comment]
its noticed that pnpm bootstrap tries to download nodejs from https://nodejs.org/download/release/v18.14.0/node-v18.14.0-linux-x64.tar.gz
image

and it often times out. Its likely these things might get rate limited by these servers
ideally we should not get into this as node and pnpm are already installed
Few things i can think of
as part of bootstrap, can we skip downloading nodejs ? do we have control ?
if its needed, then why is it not using cache?
https://github.com/nocodb/nocohub/actions/runs/6322616490/job/17168567382

other related delays in pnpm get store path as well 
image

Conclusion : 
Looks like a bug in pnpm, started the discussion with pnpm team here https://github.com/orgs/pnpm/discussions/7140 tagged the code in question in this discussion
Found a workaround for now and have a pr open.
Basically pnpm setup step deletes the nodejs directory within pnpm folder and then any pnpm install command starts downloading nodejs again.
Tried standalone mode in setup pnpm but it didn’t help.

fix/ Workaround : is to not setup pnpm if required version is already available on machine
